walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java

```diff
@@ -257,18 +291,193 @@ public void notEnoughRoom() {
+            new Event(
+                "Event 2",
+                TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TimeRange.END_OF_DAY, true),
+                Collections.singletonList(PERSON_B)));
+
+    MeetingRequest request = new MeetingRequest(Collections.emptyList(), DURATION_30_MINUTES);
+    request.addOptionalAttendee(PERSON_A);
+    request.addOptionalAttendee(PERSON_B);
+
+    Collection<TimeRange> actual = query.query(events, request);
+
+    Collection<TimeRange> expected = Collections.emptyList();
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void noMandatoryAttendeesGaps() {
+
+    // Two optional attendees with the following gaps in their schedules
+    // Should return time slots that both attendees can attend
+    //
+    // Events  :      |--A--| 
+    // Events  :            |-B-|  |-B-|
+    // Day     : |---------------------|
+    // Options : |----|     
+    // Options :                |--|
+
+      
+    Collection<Event> events =
+        Arrays.asList(
+            new Event(
+                "Event 1",
+                TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0930AM, false),
+                Collections.singletonList(PERSON_A)),
+            new Event(
+                "Event 2",
+                TimeRange.fromStartDuration(TIME_1000AM, 30),
+                Collections.singletonList(PERSON_A)),
+            new Event(
+                "Event 3",
+                TimeRange.fromStartDuration(TimeRange.getTimeInMinutes(16, 0), 30),
+                Collections.singletonList(PERSON_A)),
+            new Event(
+                "Event 4",
+                TimeRange.fromStartEnd(
+                    TimeRange.getTimeInMinutes(17, 0), TimeRange.END_OF_DAY, true),
+                Collections.singletonList(PERSON_A)),
+            new Event(
+                "Event 1.5",
+                TimeRange.fromStartDuration(TimeRange.getTimeInMinutes(9, 0), 30),
+                Collections.singletonList(PERSON_B)),
+            new Event(
+                "Event 5",
+                TimeRange.fromStartDuration(TimeRange.getTimeInMinutes(14, 0), 120),
+                Collections.singletonList(PERSON_B)));
+
+    MeetingRequest request = new MeetingRequest(Collections.emptyList(), DURATION_30_MINUTES);
+    request.addOptionalAttendee(PERSON_A);
+    request.addOptionalAttendee(PERSON_B);
+
+    Collection<TimeRange> actual = query.query(events, request);
+
+    Collection<TimeRange> expected =
+        Arrays.asList(
+            TimeRange.fromStartEnd(
+                TimeRange.getTimeInMinutes(9, 30), TimeRange.getTimeInMinutes(10, 0), false),
+            TimeRange.fromStartEnd(
+                TimeRange.getTimeInMinutes(10, 30), TimeRange.getTimeInMinutes(14, 0), false),
+            TimeRange.fromStartEnd(
+                TimeRange.getTimeInMinutes(16, 30), TimeRange.getTimeInMinutes(17, 0), false));
+    Assert.assertEquals(expected, actual);
```

Comment:
The graphic shows two possible options but three are expected here.

Reply:
Same as comment above